vnext/Microsoft.ReactNative/Fabric/ComponentViewRegistry.cpp

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "pch.h"
+
+#include "ComponentViewRegistry.h"
+
+#pragma warning(push)
+#pragma warning(disable : 4244 4305)
+#include <react/renderer/components/scrollview/ScrollViewShadowNode.h>
+#include <react/renderer/components/text/ParagraphShadowNode.h>
+#include <react/renderer/components/text/TextShadowNode.h>
+#include <react/renderer/components/view/ViewShadowNode.h>
+#pragma warning(pop)
+
+#include "ParagraphComponentView.h"
+#include "ScrollViewComponentView.h"
+#include "TextComponentView.h"
+#include "ViewComponentView.h"
+
+namespace Microsoft::ReactNative {
+
+ComponentViewDescriptor const &ComponentViewRegistry::dequeueComponentViewWithComponentHandle(
+    facebook::react::ComponentHandle componentHandle,
+    facebook::react::Tag tag) noexcept {
+  // TODO implement recycled components like core does
+
+  std::shared_ptr<BaseComponentView> view;
+
+  if (componentHandle == facebook::react::TextShadowNode::Handle()) {
+    view = std::make_shared<TextComponentView>();
+  } else if (componentHandle == facebook::react::ParagraphShadowNode::Handle()) {
+    view = std::make_shared<ParagraphComponentView>();
+  } else if (componentHandle == facebook::react::ScrollViewShadowNode::Handle()) {
+    view = std::make_shared<ScrollViewComponentView>();
+  }
+
+  else {
+    // Currently we are treating a lot of component just as normal views...
+    view = std::make_shared<ViewComponentView>();
+  }
+
+  view->Element().Tag(winrt::box_value(tag));
```

Comment:
How private is the underlying UIElement? Wondering if we should avoid the builtin tag prop.

Reply:
Yes we shouldn't be using tag.  But we are already doing so: https://github.com/microsoft/react-native-windows/issues/4187 is tracking changing that.